<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="github-pandoc.css" type="text/css" />
</head>
<body>
<h1 id="cis-eqtl-analysis-with-dap-a-step-by-step-guide">cis-eQTL analysis with DAP: a step-by-step guide</h1>
<p>This page provides a step-by-by guide to analyze cis-eQTL data (from a single tissue) using the software package DAP, which enables highly effient Bayesian multi-SNP analysis. Among many important features, DAP allows to incorporate genomic annotations into the eQTL mapping.</p>
<p>Here we show a step-by-step guide to analyze a subset of <a href="http://www.geuvadis.org/">GEUVADIS</a> data that contains genotype-expression data for 92 Toscani(TSI) samples. This data set contains expression data of 11,837 protein-coding and LincRNA genes measured in LCLs. The candidate SNPs for cis-eQTL mapping are those within the 100kb radius of the trasnscription start site (TSS) of each gene.</p>
<p>The described analysis is performed in a single multi-core Linux box. The procedure utilizes the feature of multi-thread processing. It can also be adjusted to run in a cluster environment.</p>
<h2 id="step-1-software-installation">Step 1: software installation</h2>
<p>The following binary executables are <strong><em>required</em></strong> by the analysis * <a href="https://github.com/xqwen/dap/tree/master/dap_greedy_src">dap-g</a>: the newest implementation of the DAP algorithm for multi-SNP fine-mapping * <a href="https://github.com/xqwen/dap/tree/master/torus_src">torus</a>: for prior specification</p>
<p>The following utility is recommended * <a href="https://github.com/xqwen/openmp_wrapper">openmp-wrapper</a>: for automatic multi-thread processing</p>
<p>Download and compile the source code from the above URLs and make the binary executables accessible to the analysis.</p>
<h2 id="step-2-data-preparation">Step 2: data preparation</h2>
<p>After standard QC and pre-processing steps, the genotype-phenotype information of each gene should be organized into a single text file. The file format is explained in <a href="https://github.com/xqwen/dap/wiki/Case-study:-multi-SNP-fine-mapping#genotype-phenotype-data-file-required">here</a>. The formatted genotype-phenotype data files for 11,837 genes in GEUVADIS TSI samples, <code>geuv.tsi.eqtl.sbams.tgz</code>, can be downloaded from <a href="http://www-personal.umich.edu/~xwen/download/qtl_example/geuv.tsi.eqtl.sbams.tgz">here</a>.</p>
<p>The illustrated analysis also utilizes SNP and gene position information. The SNP position file, <code>geuv.snp.map.gz</code>, can be downloaded from <a href="http://www-personal.umich.edu/~xwen/download/qtl_example/geuv.snp.map.gz">here</a>. The Gene position file, <code>geuv.gene.map.gz</code>, can be downloaded from <a href="http://www-personal.umich.edu/~xwen/download/qtl_example/geuv.gene.map.gz">here</a>.</p>
<h2 id="step-3-set-up-working-directory">Step 3: set up working directory</h2>
<ol style="list-style-type: decimal">
<li>create a working directory <code>workspace</code>, this directory is assumed to be the current working directory (cwd) from this point on.</li>
<li>create an empty directory <code>workspace/sbams_data/</code> and move the downloaded data file into this directory</li>
<li>unpack the data file: <code>cd sbsams_data; tar zxf geuv.tsi.eqtl.sbams.tgz</code>. In the end, there should be 11,837 data files named as &quot;<strong><em>gene_name</em></strong>.dat&quot; unpacked in the <code>sbams_data</code> directory.</li>
</ol>
<h2 id="step-4-prior-estimation">Step 4: prior estimation</h2>
<p>Before the multi-SNP mapping, we first set the prior for each candidate cis-SNP. In cis-eQTL mapping, it is well-known that candidate SNPs located closer to transcription start site (TSS) is more likely to be associated with the expression level of the targe gene. We will utilize this information to quantify different priors for different SNPs according to their distance to TSS (DTSS). In particular, we treat the genomic feature DTSS as a categorical annotation and use the executable<code>torus</code> for prior estimation. The statistical procedure is described in <a href="http://projecteuclid.org/euclid.aoas/1475069621">Wen 2016</a>.</p>
<h3 id="step-4.1-obtain-single-snp-association-test-statistics">Step 4.1: obtain single-SNP association test statistics</h3>
<p>If the eQTL data are already analyzed by either <a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/"><code>MatrixEQTL</code></a> or <a href="http://fastqtl.sourceforge.net/"><code>fastQTL</code></a>, the output from either software can be fed into <code>torus</code> for prior estimation.</p>
<p>Alternatively, single-SNP Bayes factor can be computed by <code>dap-g</code> using the following command</p>
<pre><code> dap-g -d gene_name.dat -scan &gt; gene_name.bf</code></pre>
<p>where <code>gene_name.dat</code> a single sbams format genotype-expression file.</p>
<p>For batch processing,</p>
<ol style="list-style-type: decimal">
<li>download <code>batch_scan.pl</code> from the <a href="https://github.com/xqwen/qtl_pipeline/tree/master/scripts">repo</a> into the <code>workspace</code> directory</li>
<li>create directory <code>workspace/scan_out</code></li>
<li>run <code>perl batch_scan.pl &gt; batch_scan.cmd</code></li>
<li>batch processing by <code>openmp_wrapper -d batch_scan.cmd -t 8</code> where &quot;-t 8&quot; specifices that 8 parallel threads are requested.</li>
<li>upon completion, obtain the combine the data by <code>cat scan_out/*.bf | gzip - &gt; geuv.tsi.bf.gz</code></li>
</ol>
<p>The output <code>guev.tsi.bf.gz</code> should appear in <code>workspace</code> upon completion.</p>
<h3 id="step-4.2-run-torus">Step 4.2: run <code>torus</code></h3>
<p>To obtain the priors, issue the following command from <code>workspace</code> directory</p>
<p><code>torus -d geuv.tsi.bf.gz -smap geuv.snp.map.gz -gmap geuv.gene.map.gz --load_bf -dump_prior priors</code></p>
<p>In the end, 11,837 prior files should be output into the newly created directory <code>workspace/priors</code>. It is important to emphasize here that prior should be estimated for <strong><em>all genes</em></strong> (instead of just a few selected ones) for the necessary statistical rigor.</p>
<h2 id="step-5-multi-snp-fine-mapping">Step 5: multi-SNP fine-mapping</h2>
<p>With the prior and genotype-phenotype file ready, the multi-SNP cis-eQTL mapping can be achieved by the following command</p>
<p><code>dap-g -d gene_name.dat -p gene_name.prior -t 4 -ld_control 0.25 --no_size_limit &gt; gene_name.fm.out</code></p>
<p>The command line options are explained in below</p>
<ul>
<li><code>-d gene_name.dat</code>: specify the sbams format genotype-phenotype data</li>
<li><code>-p gene_name.prior</code>: specify the prior file for the corresponding gene generated by <code>torus</code></li>
<li><code>-t 4</code>: running <code>dap-g</code> with 4 parallel threads</li>
<li><code>-ld_control 0.25</code>: lower r^2 threshold for <code>dap-g</code> to consider multiple SNPs (in LD) responsible to a single association signal. (if not specified, the thereshold is 0)</li>
<li><code>--no_size_limit</code> : do not restrict the number of SNPs within a single association signal cluster (the default value is set to be 25)</li>
</ul>
<p>For batch processing,</p>
<ol style="list-style-type: decimal">
<li>download <code>batch_dap.pl</code> from the <a href="https://github.com/xqwen/qtl_pipeline/tree/master/scripts">repo</a> into the <code>workspace</code> directory</li>
<li>create directory <code>workspace/dap_out</code></li>
<li>run <code>perl batch_dap.pl &gt; batch_dap.cmd</code></li>
<li>batch processing by <code>openmp_wrapper -d batch_dap.cmd -t 4</code> where &quot;-t 4&quot; specifices that 8 parallel threads are requested.</li>
</ol>
<p>The result files are output to the directory <code>dap_out</code>.</p>
<h2 id="step-6-post-analysis-utility-ties">Step 6: Post-analysis utility ties</h2>
<p>Utilities for summarizing and visualizing fine-mapping analysis results, along with their demonstrations, are provided in <a href="https://github.com/xqwen/qtl_pipeline/tree/master/utility">this repo</a>. The utilities include</p>
<ul>
<li>Signal cluster visualization: <code>plot_dap</code></li>
</ul>
</body>
</html>
